.github/workflows/generate-release.yml

```diff
@@ -0,0 +1,292 @@
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Need full history to check existing tags
+
+      - name: Validate inputs
+        id: validate
+        run: |
+          # Validate version formats and check for existing releases
+          [[ "${{ github.event.inputs.operator_version }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] || { echo "Error: Invalid operator version format"; exit 1; }
+          [[ "${{ github.event.inputs.llamastack_version }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] || { echo "Error: Invalid llamastack version format"; exit 1; }
+          git tag -l | grep -q "^v${{ github.event.inputs.operator_version }}$" && { echo "Error: Tag v${{ github.event.inputs.operator_version }} already exists"; exit 1; }
+
+          # Extract major.minor for branch name (e.g., "0.2.1" -> "0.2")
+          major_minor=$(echo "${{ github.event.inputs.operator_version }}" | cut -d. -f1-2)
+          git ls-remote --heads origin | grep -q "release-$major_minor" && { echo "Error: Branch release-$major_minor already exists"; exit 1; }
+
+          # Set outputs for later steps
+          echo "operator_version=${{ github.event.inputs.operator_version }}" >> $GITHUB_OUTPUT
+          echo "llamastack_version=${{ github.event.inputs.llamastack_version }}" >> $GITHUB_OUTPUT
+          echo "release_branch=release-$major_minor" >> $GITHUB_OUTPUT
+
+  update-files:
+    needs: validate-inputs
+    runs-on: ubuntu-24.04
+    env:
+      operator_version: ${{ needs.validate-inputs.outputs.operator_version }}
+      llamastack_version: ${{ needs.validate-inputs.outputs.llamastack_version }}
+      release_branch: ${{ needs.validate-inputs.outputs.release_branch }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Configure git
+        run: |
+          git config user.name "${{ env.GIT_USER_NAME }}"
+          git config user.email "${{ env.GIT_USER_EMAIL }}"
+
+      - name: Create and push release branch
+        run: |
+          git checkout -b "$release_branch"
+          git push origin "$release_branch"
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Prepare release files
+        run: |
+          # Use the new make release command
+          make release VERSION=$operator_version LLAMASTACK_VERSION=$llamastack_version
+
+      - name: Commit changes
+        run: |
+          git add .
+          if git diff --staged --quiet; then
+            echo "No changes to commit"
+          else
+            git commit -s -m "Release v$operator_version" \
+                         -m "- Update operator version to v$operator_version" \
+                         -m "- Update LlamaStack distributions to $llamastack_version" \
+                         -m "- Generate release manifests"
+            git push origin $release_branch
```

Comment:
We should be creating a PR here from a temporary branch. This will allow us to run the tests and all other checks as we do for other PRs and merge them only if all checks are successful.